app/controllers/budget_plans_controller.rb

```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+class BudgetPlansController < ApplicationController
+  def index
+    begin
+      @budgets = BudgetPlan.where(user_id: current_user.id)
+    rescue ActiveRecord::RecordNotFound => e
```

Comment:
this is never reached, since  `BudgetPlan.where(user_id: current_user.id)` would return empty.

Reply:
good point will re-code using a boolean expression